guides/testing.md

```diff
@@ -108,18 +113,75 @@ end
 
 They return the rescued exception so it can be used for further expectations, for example to verify specific properties of the exception.
 
+## Focusing on a group of specs
+
+A `describe`, `context` or `it` can be marked with `focus: true`, like this:
+
+```crystal
+it "adds", focus: true do
+  (2 + 2).should_not eq(5)
+end
+```
+
+If any such thing is marked with `focus: true` then only those examples will run.
+
+## Tagging specs
+
+Tags allow you to group your specs, allowing you to run a subset of your specs.
+
+A `describe`, `context` or `it` can be tagged, like this:
+
+```crystal
+it "is slow", tags: "slow" do
+  sleep 60
+  true.should be(true)
+end
+
+it "is fast", tags: "fast" do
+  true.should be(true)
+end
+```
+
+Tagging a `describe` or `context` implies that all of the containing `it`s are also tagged.
+
+Multiple tags can be specified by giving an Enumerable, such as Array or Set.
+
 ## Running specs
 
 The Crystal compiler has a `spec` command with tools to constrain which examples get run and tailor the output. All specs of a project are compiled and executed through the command `crystal spec`.
 
 By convention, specs live in the `spec/` directory of a project. Spec files must end with `_spec.cr` to be recognizable as such by the compiler command.
 
-You can compile and run specs from folder trees, individual files or specific lines in a file.
+You can compile and run specs from folder trees, individual files, or specific lines in a file. If the specified line is the beginning of a `describe` or `context` section, all specs inside that group are run.
+
+The default formatter outputs the file and line style command for failing specs which makes it easy to rerun just this individual spec.
 
 You can turn off colors with the switch `--no-color`.
 
+### Randomizing order of specs
+
+Specs, by default, run in the order defined, but can be run in a random order by passing `--order random` to `crystal spec`.
+
+Specs run in random order will display a seed value upon completion. This seed value can be used to rerun the specs in that same order by passing the seed value to `--order`.
+
+### Filtering
```

Comment:
Could you please move these changed spec runner CLI options to `using_the_compiler/README.md`?